### ISSUE TITLE
Product was indexed but is not showing up

### ISSUE THREAD

**Issue by Mobecls** [algolia/algoliasearch-magento#657]
The product was deleted from algolia. 

We have a configurable product that is enabled, has stock and visible on website. And visible via direct url. But not visible in algolia. 

We have done full reindex. In logs I can see, that this product ID was reindexed:
2016-11-03T16:32:18+00:00 DEBUG (7): >>>>> BEGIN CREATE RECORD 1875 1 (Store View)
2016-11-03T16:32:18+00:00 DEBUG (7): Product type (configurable, mapped to: configurable)
2016-11-03T16:32:19+00:00 DEBUG (7): <<<<< END CREATE RECORD 1875 1 (Store View) (1.1715772151947sec)
2016-11-03T16:32:19+00:00 DEBUG (7): <<<<< END CREATE RECORDS 1 (Store View) (1.1731839179993sec)
2016-11-03T16:32:19+00:00 DEBUG (7):
2016-11-03T16:32:19+00:00 DEBUG (7):
2016-11-03T16:32:19+00:00 DEBUG (7): >>>>> BEGIN ADD/UPDATE TO ALGOLIA
2016-11-03T16:32:20+00:00 DEBUG (7): Product IDs: 1875
2016-11-03T16:32:20+00:00 DEBUG (7): <<<<< END ADD/UPDATE TO ALGOLIA (0.14631605148315sec)

But then I can see, that the product with the same ID was added again. And it was simple product: 

2016-11-03T16:38:58+00:00 DEBUG (7): >>>>> BEGIN CREATE RECORD 11875 1 (GolfDiscount Store View)
2016-11-03T16:38:58+00:00 DEBUG (7): Product type (simple, mapped to: simple)
2016-11-03T16:38:58+00:00 DEBUG (7): <<<<< END CREATE RECORD 11875 1 (GolfDiscount Store View) (0.35840082168579sec)

And now the product is not showing in algolia (but showing on website via direct url). It also can't be find in algolia dashboard...

**Comment by JanPetr** [comment]
Hello @Mobecls,
it the logs there are two different products. One product with ID 1875 and the second one with ID 11875.
But both products should be indexed, but in different indices as the second one should be only in "GolfDiscount" store. Can you check it?

**Comment by Mobecls** [comment]
1875 - is missing... http://i.imgur.com/Ks1IUVa.png

it is available on website and I can open it via direct url, but i can't find it via search. Nor I can't find it in categories....

Maybe it is connected to our API keys issue?

**Comment by Mobecls** [comment]
API Key issue was solved, after we have done reindex 3 times and all of the logs has a record:

016-11-05T08:02:59+00:00 DEBUG (7): >>>>> BEGIN CREATE RECORD 1875 1 (GolfDiscount Store View)
2016-11-05T08:02:59+00:00 DEBUG (7): Product type (configurable, mapped to: configurable)
2016-11-05T08:03:01+00:00 DEBUG (7): <<<<< END CREATE RECORD 1875 1 (GolfDiscount Store View) (2.3191020488739sec)
2016-11-05T08:03:01+00:00 DEBUG (7): <<<<< END CREATE RECORDS 1 (GolfDiscount Store View) (2.3210890293121sec)
2016-11-05T08:03:01+00:00 DEBUG (7):
2016-11-05T08:03:01+00:00 DEBUG (7):
2016-11-05T08:03:01+00:00 DEBUG (7): >>>>> BEGIN ADD/UPDATE TO ALGOLIA
2016-11-05T08:03:01+00:00 DEBUG (7): Product IDs: 1875
2016-11-05T08:03:01+00:00 DEBUG (7): <<<<< END ADD/UPDATE TO ALGOLIA (0.14681315422058sec)


But the product can not be found in algolia via product ID...

**Comment by Mobecls** [comment]
should I send you the full log?

**Comment by Mobecls** [comment]
We got deeper and investigated the issue. We have an extension - Amasty labels. That extension adds a label to the product.  Not sure why, but when label was generated during pushing products to algolia - it recreated the same label for childs of configurable product as well. So Algolia rejected that portion of data, without adding anything to logs. 

So we have fixed that. 

Let me know, if you would need any help with algolia.
